x-pack/plugins/infra/common/alerting/metrics/alert_link.ts

```diff
@@ -10,26 +10,68 @@ import { encode } from 'rison-node';
 import { stringify } from 'query-string';
 import { ParsedTechnicalFields } from '@kbn/rule-registry-plugin/common/parse_technical_fields';
 
+export const flatAlertRuleParams = (params: {}, pKey = ''): Record<string, unknown[]> => {
+  return Object.entries(params).reduce((acc, [key, field]) => {
+    const objectKey = pKey.length ? `${pKey}.${key}` : key;
+    if (typeof field === 'object' && field != null) {
+      if (Array.isArray(field) && field.length > 0) {
+        return {
+          ...acc,
+          ...flatAlertRuleParams(field[0] as {}, objectKey),
+        };
+      } else {
+        return {
+          ...acc,
+          ...flatAlertRuleParams(field as {}, objectKey),
+        };
+      }
+    }
+    return {
+      ...acc,
+      [objectKey]: Array.isArray(field) ? field : [field],
+    };
+  }, {});
+};
+
 export const getInventoryViewInAppUrl = (
   fields: ParsedTechnicalFields & Record<string, any>
 ): string => {
+  let inventoryFields = fields;
+
+  /* Temporary Solution
```

Comment:
Might be good to get a github issue open and linked in here so we can remove the workaround eventually.